django/utils/html.py

```diff
@@ -442,7 +443,7 @@ def trim_punctuation(self, word):
                     rstripped = middle.rstrip(closing)
                     if rstripped != middle:
                         strip = counts[closing] - counts[opening]
-                        trail = middle[-strip:]
+                        trail.appendleft(middle[-strip:])
```

Comment:
This seems like a wrong transformation - we don't want to append, but fully replace…? So more like:

```python
trail.clear()
trail.append(middle[-strip:])
```